File: problems/791_custom_sort_string/Solution.java
import java.util.*;

class Solution {
    public String customSortString(String order, String s) {
        int[] rank = new int[26];
        Arrays.fill(rank, 26);
        for (int i = 0; i < order.length(); i++) rank[order.charAt(i) - 'a'] = i;
        Character[] chars = new Character[s.length()];
        for (int i = 0; i < s.length(); i++) chars[i] = s.charAt(i);
        Arrays.sort(chars, (a, b) -> rank[a - 'a'] - rank[b - 'a']);
        StringBuilder sb = new StringBuilder();
        for (char c : chars) sb.append(c);
        return sb.toString();
    }
}

File: problems/791_custom_sort_string/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    string customSortString(string order, string s) {
        int rank[26];
        fill(rank, rank + 26, 26);
        for (int i = 0; i < (int)order.size(); i++) rank[order[i] - 'a'] = i;
        sort(s.begin(), s.end(), [&](char a, char b) { return rank[a - 'a'] < rank[b - 'a']; });
        return s;
    }
};
